### PR TITLE
Fixes artwork revolver and rejuv interacting poorly with mods

### PR DESCRIPTION
**Pull request by vode-code**
## About The Pull Request
This PR fixes the bug where adding modifiers to an artwork revolver would change its stats to the non-randomized default and the bug where you could stack modifiers with rejuvenating fluid.

## Why It's Good For The Game
Bug Bad.

## Changelog
:cl: Chickenish
fix: Rejuv now removes modifiers from tools and guns instead of invisibly keeping them
fix: Artwork Revolvers now interact with modifiers properly
/:cl:

### CODE REVIEW

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -836,6 +837,10 @@
 	sharp = initial(sharp)
 	attack_verb = list()
 	one_hand_penalty = initial(one_hand_penalty)
+	if (custom_default.len) // this override is used by the artwork_revolver for RNG gun stats
+		for(var/propname in custom_default) // taken from gun_firemode.dm
+			if(propname in src.vars)
+				src.vars[propname] = custom_default[propname]
```
Comment (hyperioo) [inline code/modules/projectiles/gun.dm:843]:
```suggestion
			if(propname in vars)
				vars[propname] = custom_default[propname]
```
I think you don't need the src.

File: code/modules/scrap/oldificator.dm [hunk 2]
```diff
@@ -32,6 +32,37 @@
 		return TRUE
 	return FALSE
 
+/obj/item/gun/make_young()
+	var/list/stored_upgrades = item_upgrades.Copy()
+	for (var/obj/item/toremove in stored_upgrades)
```
Comment (hyperioo) [inline code/modules/scrap/oldificator.dm:37]:
```suggestion
	for(var/obj/item/toremove in stored_upgrades)
```

File: code/modules/scrap/oldificator.dm [hunk 3]
```diff
@@ -32,6 +32,37 @@
...
+		var/datum/component/item_upgrade/IU = toremove.GetComponent(/datum/component/item_upgrade)
+		if (IU)
```
Comment (hyperioo) [inline code/modules/scrap/oldificator.dm:39]:
```suggestion
		if(IU)
```

File: code/modules/scrap/oldificator.dm [hunk 4]
```diff
@@ -32,6 +32,37 @@
...
+			SEND_SIGNAL(toremove, COMSIG_REMOVE, src)
+			visible_message(SPAN_NOTICE("[toremove] detaches from [src]."))
```
Comment (hyperioo) [inline code/modules/scrap/oldificator.dm:41]:
```suggestion
			visible_message(SPAN_NOTICE("\The [toremove] detaches from \the [src]."))
```

File: code/modules/scrap/oldificator.dm [hunk 5]
```diff
@@ -32,6 +32,37 @@
...
+			. = TRUE
+
+	refresh_upgrades()
+	if (.) // this is so it always returns true if it did something
```
Comment (hyperioo) [inline code/modules/scrap/oldificator.dm:45]:
```suggestion
	if(.) // this is so it always returns true if it did something
```

File: code/modules/scrap/oldificator.dm [hunk 6]
```diff
@@ -32,6 +32,37 @@
...
+		..()
+	else
+		. = ..()
+
+/obj/item/tool/make_young()
+	var/list/stored_upgrades = item_upgrades.Copy()
+	for (var/obj/item/toremove in stored_upgrades)
```
Comment (hyperioo) [inline code/modules/scrap/oldificator.dm:52]:
```suggestion
	for(var/obj/item/toremove in stored_upgrades)
```

File: code/modules/scrap/oldificator.dm [hunk 7]
```diff
@@ -32,6 +32,37 @@
...
+		var/datum/component/item_upgrade/IU = toremove.GetComponent(/datum/component/item_upgrade)
+		if (IU)
```
Comment (hyperioo) [inline code/modules/scrap/oldificator.dm:54]:
```suggestion
		if(IU)
```

File: code/modules/scrap/oldificator.dm [hunk 8]
```diff
@@ -32,6 +32,37 @@
...
+			SEND_SIGNAL(toremove, COMSIG_REMOVE, src)
+			visible_message(SPAN_NOTICE("[toremove] detaches from [src]."))
```
Comment (hyperioo) [inline code/modules/scrap/oldificator.dm:56]:
```suggestion
			visible_message(SPAN_NOTICE("\The [toremove] detaches from \the [src]."))
```

File: code/modules/scrap/oldificator.dm [hunk 9]
```diff
@@ -32,6 +32,37 @@
...
+			. = TRUE
+
+	refresh_upgrades()
+	if (.) // this is so it always returns true if it did something
```
Comment (hyperioo) [inline code/modules/scrap/oldificator.dm:60]:
```suggestion
	if(.) // this is so it always returns true if it did something
```

**Review by hyperioo — CHANGES_REQUESTED** [review]

**Review by TheShown911 — COMMENTED** [review]
Code looks fine, but merge conflict.

File: code/modules/projectiles/gun.dm [hunk 10]
```diff
@@ -836,6 +837,10 @@
 	sharp = initial(sharp)
 	attack_verb = list()
 	one_hand_penalty = initial(one_hand_penalty)
+	if (custom_default.len) // this override is used by the artwork_revolver for RNG gun stats
```
[the remark this answers is not in the document]
Comment (vode-code) [reply code/modules/projectiles/gun.dm:840]: unnecessary stylistic change

File: code/modules/scrap/oldificator.dm (hunk 2 above)
Comment (vode-code) [reply code/modules/scrap/oldificator.dm:37]: unnecessary stylistic change

File: code/modules/scrap/oldificator.dm (hunk 3 above)
Comment (vode-code) [reply code/modules/scrap/oldificator.dm:39]: unnecessary stylistic change

File: code/modules/scrap/oldificator.dm (hunk 5 above)
Comment (vode-code) [reply code/modules/scrap/oldificator.dm:45]: unnecessary stylistic change

File: code/modules/scrap/oldificator.dm (hunk 6 above)
Comment (vode-code) [reply code/modules/scrap/oldificator.dm:52]: unnecessary stylistic change

File: code/modules/scrap/oldificator.dm (hunk 7 above)
Comment (vode-code) [reply code/modules/scrap/oldificator.dm:54]: unnecessary stylistic change

File: code/modules/scrap/oldificator.dm (hunk 9 above)
Comment (vode-code) [reply code/modules/scrap/oldificator.dm:60]: unnecessary stylistic change

**Comment by vode-code** [comment]
should no longer be merge conflicting, not sure why it still has the tag.

**Review by hyperioo — COMMENTED** [review]
Code is ok even if style suggestion have been discarded

**Comment by hyperioo** [comment]
Conflict because of `code/game/machinery/autolathe/artist_bench.dm` according to the git interface.

**Review by drexample — CHANGES_REQUESTED** [review]
Breh it don't compile

**Comment by vode-code** [comment]
I have fixed the weird half-update bugs causing compile errors by fully updating.

**Pull request closed without merging**